crates/bitwarden-uuid-macro/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, Ident, Visibility,
+};
+
+#[proc_macro]
+pub fn uuid(input: TokenStream) -> TokenStream {
+    // Parse input as: vis ident
+    let input = parse_macro_input!(input as IdTypeInput);
+    let ident = input.ident;
+    let vis = input.vis;
+    let name_str = ident.to_string();
+
+    let tsify_type = format!("Tagged<Uuid, \"{}\">", name_str);
+
+    let expanded = quote! {
+        #[cfg(feature = "wasm")]
+        #[derive(serde::Serialize, serde::Deserialize, tsify_next::Tsify)]
+        #[tsify(into_wasm_abi, from_wasm_abi)]
+        #[repr(transparent)]
+        #vis struct #ident(
+            #[tsify(type = #tsify_type)]
+            uuid::Uuid
+        );
+
+        #[cfg(not(feature = "wasm"))]
+        #[derive(serde::Serialize, serde::Deserialize)]
+        #[repr(transparent)]
+        #vis struct #ident(
+            uuid::Uuid
+        );
+
+        #[cfg(feature = "uniffi")]
+        uniffi::custom_newtype!(#ident, uuid::Uuid);
+
+        impl #ident {
+            pub fn new(value: uuid::Uuid) -> Self {
+                Self(value)
+            }
+        }
+
+        impl std::str::FromStr for #ident {
```

Comment:
suggestion: not sure if someone would rename `std` but 🤷  
```suggestion
        impl ::std::str::FromStr for #ident {
```